### PR TITLE
Update portcullis to 1.2.3

### PR DESCRIPTION
**Pull request by ljyanesm**

### CODE REVIEW

File: var/spack/repos/builtin/packages/portcullis/package.py [hunk 1]
```diff
@@ -10,9 +10,10 @@
 class Portcullis(AutotoolsPackage):
     """PORTable CULLing of Invalid Splice junctions"""
 
-    homepage = "https://github.com/maplesond/portcullis"
-    url      = "https://github.com/maplesond/portcullis/archive/Release-1.1.2.tar.gz"
+    homepage = "https://github.com/ei-corebioinformatics/portcullis"
+    url      = "https://github.com/ei-corebioinformatics/archive/refs/tags/Release-1.2.3.tar.gz"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/portcullis/package.py]: The `url` seems incorrect. The correct one has the form:
```
https://github.com/EI-CoreBioinformatics/portcullis/archive/refs/tags/Release-1.1.2.tar.gz
```

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/portcullis/package.py [hunk 2]
```diff
@@ -10,14 +10,16 @@
...
+    homepage = "https://github.com/EI-CoreBioinformatics/portcullis"
+    url      = "https://github.com/EI-CoreBioinformatics/portcullis/archive/refs/tags/Release-1.1.2.tar.gz"
 
+    version('1.2.3', sha256='172452b5cef12a8dcc2c1c68527000743114136ee63a0dbe307ac4e2a816bc99')
     version('1.1.2', sha256='5c581a7f827ffeecfe68107b7fe27ed60108325fd2f86a79d93f61b328687749')
 
     depends_on('autoconf@2.53:', type='build')
     depends_on('automake@1.11:', type='build')
     depends_on('libtool@2.4.2:',  type='build')
+    depends_on('boost', type='build')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/portcullis/package.py]:
```suggestion
    depends_on('boost')
```
In our current dependency model header only libraries are still of `link` type. Roughly `build` and `run` identify dependencies on executables that are used in a separate process at either build-time or run-time.

**Review by alalazo — CHANGES_REQUESTED** [review]
Only a minor request

**Comment by ljyanesm** [comment]
@alalazo, thank you for reviewing these changes

**Review by alalazo — APPROVED** [review]
Thanks!

**Pull request merged**